### PR TITLE
writing correct Tar UTF8 filenames

### PR DESCRIPTION
**Pull request by shu66y**
We had the problem of writing UTF8 Filenames within a Tar correctly. I didn't consider reading so far.

_I certify that I own, and have sufficient rights to contribute, all source code and related material intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License._

### CODE REVIEW

**Comment by piksel** [comment]
I have been going through how the tar headers are parsed for #121, and UTF-8 encoded names is probably better. The specification suggests using extended headers for encodings other than ASCII, but GNU Tar writes the UTF8-encoded bytes as the `Name` so I guess we should too.

File: src/ICSharpCode.SharpZipLib/Tar/TarHeader.cs [hunk 1]
```diff
@@ -870,9 +870,11 @@ public static int GetNameBytes(string name, int nameOffset, byte[] buffer, int b
 
 			int i;
 
-			for (i = 0; i < length && nameOffset + i < name.Length; ++i)
+			byte[] nameBytes = Encoding.UTF8.GetBytes(name);
```
Comment (piksel) [inline src/ICSharpCode.SharpZipLib/Tar/TarHeader.cs]: I doubt we want BOMs in the name

```suggestion
			byte[] nameBytes = new Encoding.UTF8Encoding(false).GetBytes(name);
```

**Comment by itn3000** [comment]
This PR's may cut the filename which has long length(at least 100 bytes in UTF-8) + non-ascii character because char above U+80 is not one byte in UTF-8.

**Comment by Numpsy** [comment]
Is this obsoleted by #364 being merged?

**Comment by piksel** [comment]
Indeed. Thanks.

**Pull request closed without merging**